### ISSUE TITLE
configure: error: Aegisub requires GL support.

### ISSUE THREAD

**Issue by kimown** [Aegisub/Aegisub#188]
I want to [compile](https://github.com/Aegisub/Aegisub/pull/165/files?short_path=04c6e90#diff-04c6e90faac2675aa89e2176d2eec7d8) Aegisub in docker container without gl environment, but when I run `./configure`, it throw error: Aegisub requires GL support, I tried `./configure --with-gl=x` but doesn't work.

**Comment by wangqr** [comment]
Can you provide the output of `./configure` and the `config.log` file, and the docker image you are using?

**Comment by kimown** [comment]
> Can you provide the output of `./configure` and the `config.log` file, and the docker image you are using?

When I try to reproduce the error, it seems not happen again and I don't know why, but then I run `make -j 4`, new error occured, it seems something wrong about luajit.
```
make[1]: Leaving directory '/opt/tiger/scripts/Aegisub/vendor/luajit/src'
g++ -MMD -MP -I/opt/tiger/scripts/Aegisub/ -I/opt/tiger/scripts/Aegisub/libaegisub/include -I/usr/include/x86_64-linux-gnu  -Wall -Wextra -Wno-unused-parameter -fno-strict-aliasing -pipe -g -std=c++11 -Wno-c++11-narrowing -Wno-unused-local-typedefs -O3  -include acconf.h  -c -o /opt/tiger/scripts/Aegisub/tools/repack-thes-dict.o /opt/tiger/scripts/Aegisub/tools/repack-thes-dict.cpp
Makefile.target:100: recipe for target '/opt/tiger/scripts/Aegisub/src/libresrc/libresrc.o' failed
make: *** [/opt/tiger/scripts/Aegisub/src/libresrc/libresrc.o] Error 1
make: *** Waiting for unfinished jobs....
```


<details>
<summary>./configure</summary>

<div>
checking for gcc... gcc<br/>checking whether the C compiler works... yes<br/>checking for C compiler default output file name... a.out<br/>checking for suffix of executables... <br/>checking whether we are cross compiling... no<br/>checking for suffix of object files... o<br/>checking whether we are using the GNU C compiler... yes<br/>checking whether gcc accepts -g... yes<br/>checking for gcc option to accept ISO C89... none needed<br/>checking how to run the C preprocessor... gcc -E<br/>checking for grep that handles long lines and -e... /bin/grep<br/>checking for egrep... /bin/grep -E<br/>checking for ANSI C header files... yes<br/>checking for sys/types.h... yes<br/>checking for sys/stat.h... yes<br/>checking for stdlib.h... yes<br/>checking for string.h... yes<br/>checking for memory.h... yes<br/>checking for strings.h... yes<br/>checking for inttypes.h... yes<br/>checking for stdint.h... yes<br/>checking for unistd.h... yes<br/>checking minix/config.h usability... no<br/>checking minix/config.h presence... no<br/>checking for minix/config.h... no<br/>checking whether it is safe to define __EXTENSIONS__... yes<br/>checking build system type... x86_64-pc-linux-gnu<br/>checking host system type... x86_64-pc-linux-gnu<br/>checking whether BUILD_CREDIT has been set... no<br/>checking for GNU make... make<br/>checking for gcc... (cached) gcc<br/>checking whether we are using the GNU C compiler... (cached) yes<br/>checking whether gcc accepts -g... (cached) yes<br/>checking for gcc option to accept ISO C89... (cached) none needed<br/>checking for g++... g++<br/>checking whether we are using the GNU C++ compiler... yes<br/>checking whether g++ accepts -g... yes<br/>checking for a BSD-compatible install... /usr/bin/install -c<br/>checking for ranlib... ranlib<br/>checking for pkg-config... /usr/bin/pkg-config<br/>checking pkg-config is at least version 0.20... yes<br/>checking for version... 9010-master-6f546951b from git<br/>checking for build date... 2020-07-18 03:09 UTC<br/>checking if compiler needs -Werror to reject unknown flags... no<br/>checking for the pthreads library -lpthreads... no<br/>checking whether pthreads work without any flags... no<br/>checking whether pthreads work with -Kthread... no<br/>checking whether pthreads work with -kthread... no<br/>checking for the pthreads library -llthread... no<br/>checking whether pthreads work with -pthread... yes<br/>checking for joinable pthread attribute... PTHREAD_CREATE_JOINABLE<br/>checking if more special flags are required for pthreads... no<br/>checking for PTHREAD_PRIO_INHERIT... yes<br/>checking whether g++ supports -std=c++11... yes<br/>checking whether g++ supports -Wno-c++11-narrowing... yes<br/>checking whether gcc supports -Wno-unused-local-typedefs... yes<br/>checking whether g++ supports -Wno-unused-local-typedefs... yes<br/>checking whether g++ supports -stdlib=libc++... no<br/>checking for main in -lm... yes<br/>checking for dlclose in -ldl... yes<br/>checking how to run the C++ preprocessor... g++ -E<br/>checking sys/time.h usability... yes<br/>checking sys/time.h presence... yes<br/>checking for sys/time.h... yes<br/>checking for X... libraries , headers <br/>checking for gethostbyname... yes<br/>checking for connect... yes<br/>checking for remove... yes<br/>checking for shmat... yes<br/>checking for IceConnectionNumber in -lICE... yes<br/>checking wether we should prefer mesa for opengl implementation... yes<br/>checking for a working OpenGL order detection... mesagl gl<br/>checking for a working OpenGL implementation by pkg-config... yes<br/>checking for GL... no<br/>checking for GL... yes<br/>checking for windows.h... no<br/>checking for GL/gl.h... yes<br/>checking for OpenGL headers... yes<br/>checking for compiling a minimal OpenGL program... yes<br/>checking for linking a minimal OpenGL program... yes<br/>checking for a working OpenGL implementation... yes<br/>checking for CFLAGS needed for OpenGL... "-I/usr/include/libdrm "<br/>checking for LIBS needed for OpenGL... "-lGL "<br/>checking for LDFLAGS needed for OpenGL... ""<br/>checking for FREETYPE... yes<br/>checking for FONTCONFIG... yes<br/>checking for LIBASS... yes<br/>checking for boostlib >= 1.50.0 (105000)... yes<br/>checking whether the Boost::Chrono library is available... yes<br/>checking for exit in -lboost_chrono... yes<br/>checking whether the Boost::Filesystem library is available... yes<br/>checking for exit in -lboost_filesystem... yes<br/>checking whether the Boost::Locale library is available... yes<br/>checking for exit in -lboost_locale... yes<br/>checking whether the Boost::Regex library is available... yes<br/>checking for exit in -lboost_regex... yes<br/>checking whether the Boost::System library is available... yes<br/>checking for exit in -lboost_system... yes<br/>checking whether the Boost::Thread library is available... yes<br/>checking for exit in -lboost_thread... yes<br/>checking boost/spirit/home/phoenix/version.hpp usability... yes<br/>checking boost/spirit/home/phoenix/version.hpp presence... yes<br/>checking for boost/spirit/home/phoenix/version.hpp... yes<br/>checking for ICU_UC... yes<br/>checking for ICU_I18N... yes<br/>checking whether boost ICU support works... yes<br/>checking for library containing iconv_open... none required<br/>checking whether iconv works... yes<br/>checking whether iconv (const) works... no<br/>checking for snd_pcm_open in -lasound... no<br/>checking for OPENAL... no<br/>checking soundcard.h usability... no<br/>checking soundcard.h presence... no<br/>checking for soundcard.h... no<br/>checking sys/soundcard.h usability... yes<br/>checking sys/soundcard.h presence... yes<br/>checking for sys/soundcard.h... yes<br/>checking for LIBPULSE... no<br/>checking for PORTAUDIO... no<br/>checking for FFMS2... yes<br/>checking for FFTW3... yes<br/>checking for HUNSPELL... no<br/>checking for UCHARDET... yes<br/>checking for LUAJIT... yes<br/>checking whether LuaJIT is compiled in 5.2 mode... no<br/>checking for lua... lua<br/>checking whether to turn on debugging... no<br/>checking whether to use exception handling in debug mode... no<br/>checking for the --enable-debug option... no<br/>checking for wx-config... /usr/bin/wx-config<br/>checking for wxWidgets version >= 3.0.0... yes (version 3.0.2)<br/>checking for wxWidgets static library... no<br/>checking wx/glcanvas.h usability... no<br/>checking wx/glcanvas.h presence... no<br/>checking for wx/glcanvas.h... no<br/>checking wx/stc/stc.h usability... no<br/>checking wx/stc/stc.h presence... no<br/>checking for wx/stc/stc.h... no<br/>checking whether to use precompiled headers... yes<br/>checking whether g++ supports -Winvalid-pch... yes<br/>checking whether g++ supports -fpch-deps... yes<br/>checking whether g++ supports -fpch-preprocess... yes<br/>checking whether make sets $(MAKE)... yes<br/>checking for a thread-safe mkdir -p... /bin/mkdir -p<br/>checking for a sed that does not truncate output... /bin/sed<br/>checking whether NLS is requested... yes<br/>checking for msgfmt... /usr/bin/msgfmt<br/>checking for gmsgfmt... /usr/bin/msgfmt<br/>checking for xgettext... /usr/bin/xgettext<br/>checking for msgmerge... /usr/bin/msgmerge<br/>checking for ld used by gcc... /usr/bin/ld<br/>checking if the linker (/usr/bin/ld) is GNU ld... yes<br/>checking for shared library run path origin... /bin/bash: ./config.rpath: No such file or directory<br/>done<br/>checking for CFPreferencesCopyAppValue... no<br/>checking for CFLocaleCopyCurrent... no<br/>checking for GNU gettext in libc... yes<br/>checking whether to use NLS... yes<br/>checking where the gettext function comes from... libc<br/>checking whether to enable the update checker... yes<br/>checking for update checker server... updates.aegisub.org<br/>checking for update checker base URL... /trunk<br/>configure: creating ./config.status<br/>config.status: creating packages/desktop/aegisub.desktop.template<br/>config.status: creating packages/desktop/aegisub.appdata.xml.template<br/>config.status: creating src/libresrc/default_config_platform.json<br/>config.status: creating tools/osx-bundle.sed<br/>config.status: creating Makefile.inc<br/>config.status: creating acconf.h<br/>config.status: executing po-directories commands<br/><br/>Configure settings<br/>  Install prefix:        /usr/local<br/>  Revision:              9010-master-6f546951b<br/>  Debug:                 no<br/>  AppImage:              no<br/>  CFLAGS:                 -Wall -Wextra -Wno-unused-parameter -std=gnu99 -pipe -g -Wno-unused-local-typedefs -O3<br/>  CXXFLAGS:               -Wall -Wextra -Wno-unused-parameter -fno-strict-aliasing -pipe -g -std=c++11 -Wno-c++11-narrowing -Wno-unused-local-typedefs -O3<br/>  CPPFLAGS:              <br/>  LDFLAGS:               <br/>  LIBS:                  -ldl -lm <br/><br/>Default Settings<br/>  Audio Player:          OSS<br/><br/>Audio Players<br/>  ALSA:                  no <br/>  OpenAL:                no <br/>  OSS:                   yes <br/>  PortAudio:             no <br/>  PulseAudio:            no <br/><br/>Misc Packages<br/>  uchardet:              yes <br/>  Hunspell:              no <br/>  FFTW3:                 yes <br/>  LuaJIT:                bundled
<div>
</details>

here is the config.log file, [config.log.txt](https://github.com/Aegisub/Aegisub/files/4941008/config.log.txt) , sorry I can't provide the docker image because it is provided by a VPS, but I can provide all my commands.
```
cat /etc/issue
Debian GNU/Linux 8 \n 

git clone https://github.com/Aegisub/Aegisub.git
cd Aegisub
apt install -y \
  autoconf libtool automake \
  libfreetype6-dev libfontconfig1-dev libass-dev \
  libboost-dev libboost-chrono-dev libboost-filesystem-dev \
  libboost-locale-dev libboost-regex-dev libboost-thread-dev \
  libffms2-dev libuchardet-dev libfftw3-dev \
  libice-dev \
  ffmpeg intltool
apt install libwxgtk3.0-dev libgtk-3-dev  -y
apt install lua5.1 libluajit-5.1-dev -y
autoreconf
./configure
make -j 4
```

**Comment by wangqr** [comment]
> `Makefile.target:100: recipe for target '/opt/tiger/scripts/Aegisub/src/libresrc/libresrc.o' failed`

The error occured when compiling libresrc.o, you need to scroll up and find the actual error when compiling that file.

**Comment by kimown** [comment]
> > `Makefile.target:100: recipe for target '/opt/tiger/scripts/Aegisub/src/libresrc/libresrc.o' failed`
> 
> The error occured when compiling libresrc.o, you need to scroll up and find the actual error when compiling that file.

<details>
<summary>error: no matching function for call to ‘wxBitmap::wxBitmap(wxImage, int, double&)’</summary>
<div>
g++ -MMD -MP -I/opt/tiger/scripts/Aegisub/src/ -I.. -I/opt/tiger/scripts/Aegisub/src/include -I/opt/tiger/scripts/Aegisub/libaegisub/include -I/opt/tiger/scripts/Aegisub/build -pthread  -I/usr/include/x86_64-linux-gnu  -pthread -I/usr/include -Wall -Wextra -Wno-unused-parameter -fno-strict-aliasing -pipe -g -std=c++11 -Wno-c++11-narrowing -Wno-unused-local-typedefs -O3 -I/usr/lib/x86_64-linux-gnu/wx/include/gtk2-unicode-3.0 -I/usr/include/wx-3.0 -D_FILE_OFFSET_BITS=64 -DWXUSINGDLL -D__WXGTK__ -pthread -include /opt/tiger/scripts/Aegisub/src/agi_pre.h  -I/usr/include/x86_64-linux-gnu  -c -o /opt/tiger/scripts/Aegisub/src/video_provider_ffmpegsource.o /opt/tiger/scripts/Aegisub/src/video_provider_ffmpegsource.cpp<br/>/opt/tiger/scripts/Aegisub/src/libresrc/libresrc.cpp: In function ‘wxBitmap libresrc_getimage(const unsigned char*, size_t, double, int)’:<br/>/opt/tiger/scripts/Aegisub/src/libresrc/libresrc.cpp:26:42: error: no matching function for call to ‘wxBitmap::wxBitmap(wxImage, int, double&)’<br/>   return wxBitmap(wxImage(mem), -1, scale);<br/>                                          ^<br/>/opt/tiger/scripts/Aegisub/src/libresrc/libresrc.cpp:26:42: note: candidates are:<br/>In file included from /usr/include/wx-3.0/wx/bitmap.h:277:0,<br/>                 from /opt/tiger/scripts/Aegisub/src/libresrc/libresrc.cpp:17:<br/>/usr/include/wx-3.0/wx/gtk/bitmap.h:137:5: note: wxBitmap::wxBitmap(GdkPixmap*)<br/>     wxBitmap(GdkPixmap* pixmap);<br/>     ^<br/>/usr/include/wx-3.0/wx/gtk/bitmap.h:137:5: note:   candidate expects 1 argument, 3 provided<br/>/usr/include/wx-3.0/wx/gtk/bitmap.h:83:5: note: wxBitmap::wxBitmap(GdkPixbuf*, int)<br/>     wxBitmap(GdkPixbuf* pixbuf, int depth = 0);<br/>     ^<br/>/usr/include/wx-3.0/wx/gtk/bitmap.h:83:5: note:   candidate expects 2 arguments, 3 provided<br/>/usr/include/wx-3.0/wx/gtk/bitmap.h:81:5: note: wxBitmap::wxBitmap(const wxImage&, int)<br/>     wxBitmap(const wxImage& image, int depth = wxBITMAP_SCREEN_DEPTH);<br/>     ^<br/>/usr/include/wx-3.0/wx/gtk/bitmap.h:81:5: note:   candidate expects 2 arguments, 3 provided<br/>/usr/include/wx-3.0/wx/gtk/bitmap.h:79:5: note: wxBitmap::wxBitmap(const wxString&, wxBitmapType)<br/>     wxBitmap( const wxString &filename, wxBitmapType type = wxBITMAP_DEFAULT_TYPE );<br/>     ^<br/>/usr/include/wx-3.0/wx/gtk/bitmap.h:79:5: note:   candidate expects 2 arguments, 3 provided<br/>/usr/include/wx-3.0/wx/gtk/bitmap.h:73:5: note: wxBitmap::wxBitmap(const char* const*)<br/>     wxBitmap( const char* const* bits );<br/>     ^<br/>/usr/include/wx-3.0/wx/gtk/bitmap.h:73:5: note:   candidate expects 1 argument, 3 provided<br/>/usr/include/wx-3.0/wx/gtk/bitmap.h:72:5: note: wxBitmap::wxBitmap(const char*, int, int, int)<br/>     wxBitmap( const char bits[], int width, int height, int depth = 1 );<br/>     ^<br/>/usr/include/wx-3.0/wx/gtk/bitmap.h:72:5: note:   no known conversion for argument 1 from ‘wxImage’ to ‘const char*’<br/>/usr/include/wx-3.0/wx/gtk/bitmap.h:70:5: note: wxBitmap::wxBitmap(const wxSize&, int)<br/>     wxBitmap( const wxSize& sz, int depth = wxBITMAP_SCREEN_DEPTH )<br/>     ^<br/>/usr/include/wx-3.0/wx/gtk/bitmap.h:70:5: note:   candidate expects 2 arguments, 3 provided<br/>/usr/include/wx-3.0/wx/gtk/bitmap.h:68:5: note: wxBitmap::wxBitmap(int, int, int)<br/>     wxBitmap( int width, int height, int depth = wxBITMAP_SCREEN_DEPTH )<br/>     ^<br/>/usr/include/wx-3.0/wx/gtk/bitmap.h:68:5: note:   no known conversion for argument 1 from ‘wxImage’ to ‘int’<br/>/usr/include/wx-3.0/wx/gtk/bitmap.h:67:5: note: wxBitmap::wxBitmap()<br/>     wxBitmap() { }<br/>     ^<br/>/usr/include/wx-3.0/wx/gtk/bitmap.h:67:5: note:   candidate expects 0 arguments, 3 provided<br/>/usr/include/wx-3.0/wx/gtk/bitmap.h:64:24: note: wxBitmap::wxBitmap(const wxBitmap&)<br/> class WXDLLIMPEXP_CORE wxBitmap: public wxBitmapBase<br/>                        ^<br/>/usr/include/wx-3.0/wx/gtk/bitmap.h:64:24: note:   candidate expects 1 argument, 3 provided<br/>/opt/tiger/scripts/Aegisub/src/libresrc/libresrc.cpp:27:50: error: no matching function for call to ‘wxBitmap::wxBitmap(wxImage, int, double&)’<br/>  return wxBitmap(wxImage(mem).Mirror(), -1, scale);<br/>                                                  ^<br/>/opt/tiger/scripts/Aegisub/src/libresrc/libresrc.cpp:27:50: note: candidates are:<br/>In file included from /usr/include/wx-3.0/wx/bitmap.h:277:0,<br/>                 from /opt/tiger/scripts/Aegisub/src/libresrc/libresrc.cpp:17:<br/>/usr/include/wx-3.0/wx/gtk/bitmap.h:137:5: note: wxBitmap::wxBitmap(GdkPixmap*)<br/>     wxBitmap(GdkPixmap* pixmap);<br/>     ^<br/>/usr/include/wx-3.0/wx/gtk/bitmap.h:137:5: note:   candidate expects 1 argument, 3 provided<br/>/usr/include/wx-3.0/wx/gtk/bitmap.h:83:5: note: wxBitmap::wxBitmap(GdkPixbuf*, int)<br/>     wxBitmap(GdkPixbuf* pixbuf, int depth = 0);<br/>     ^<br/>/usr/include/wx-3.0/wx/gtk/bitmap.h:83:5: note:   candidate expects 2 arguments, 3 provided<br/>/usr/include/wx-3.0/wx/gtk/bitmap.h:81:5: note: wxBitmap::wxBitmap(const wxImage&, int)<br/>     wxBitmap(const wxImage& image, int depth = wxBITMAP_SCREEN_DEPTH);<br/>     ^<br/>/usr/include/wx-3.0/wx/gtk/bitmap.h:81:5: note:   candidate expects 2 arguments, 3 provided<br/>/usr/include/wx-3.0/wx/gtk/bitmap.h:79:5: note: wxBitmap::wxBitmap(const wxString&, wxBitmapType)<br/>     wxBitmap( const wxString &filename, wxBitmapType type = wxBITMAP_DEFAULT_TYPE );<br/>     ^<br/>/usr/include/wx-3.0/wx/gtk/bitmap.h:79:5: note:   candidate expects 2 arguments, 3 provided<br/>/usr/include/wx-3.0/wx/gtk/bitmap.h:73:5: note: wxBitmap::wxBitmap(const char* const*)<br/>     wxBitmap( const char* const* bits );<br/>     ^<br/>/usr/include/wx-3.0/wx/gtk/bitmap.h:73:5: note:   candidate expects 1 argument, 3 provided<br/>/usr/include/wx-3.0/wx/gtk/bitmap.h:72:5: note: wxBitmap::wxBitmap(const char*, int, int, int)<br/>     wxBitmap( const char bits[], int width, int height, int depth = 1 );<br/>     ^<br/>/usr/include/wx-3.0/wx/gtk/bitmap.h:72:5: note:   no known conversion for argument 1 from ‘wxImage’ to ‘const char*’<br/>/usr/include/wx-3.0/wx/gtk/bitmap.h:70:5: note: wxBitmap::wxBitmap(const wxSize&, int)<br/>     wxBitmap( const wxSize& sz, int depth = wxBITMAP_SCREEN_DEPTH )<br/>     ^<br/>/usr/include/wx-3.0/wx/gtk/bitmap.h:70:5: note:   candidate expects 2 arguments, 3 provided<br/>/usr/include/wx-3.0/wx/gtk/bitmap.h:68:5: note: wxBitmap::wxBitmap(int, int, int)<br/>     wxBitmap( int width, int height, int depth = wxBITMAP_SCREEN_DEPTH )<br/>     ^<br/>/usr/include/wx-3.0/wx/gtk/bitmap.h:68:5: note:   no known conversion for argument 1 from ‘wxImage’ to ‘int’<br/>/usr/include/wx-3.0/wx/gtk/bitmap.h:67:5: note: wxBitmap::wxBitmap()<br/>     wxBitmap() { }<br/>     ^<br/>/usr/include/wx-3.0/wx/gtk/bitmap.h:67:5: note:   candidate expects 0 arguments, 3 provided<br/>/usr/include/wx-3.0/wx/gtk/bitmap.h:64:24: note: wxBitmap::wxBitmap(const wxBitmap&)<br/> class WXDLLIMPEXP_CORE wxBitmap: public wxBitmapBase<br/>                        ^<br/>/usr/include/wx-3.0/wx/gtk/bitmap.h:64:24: note:   candidate expects 1 argument, 3 provided<br/>/opt/tiger/scripts/Aegisub/src/libresrc/libresrc.cpp:28:1: warning: control reaches end of non-void function [-Wreturn-type]<br/> }<br/> ^<br/>At global scope:<br/>cc1plus: warning: unrecognized command line option "-Wno-c++11-narrowing"<br/>Makefile.target:100: recipe for target '/opt/tiger/scripts/Aegisub/src/libresrc/libresrc.o' failed<br/>make: *** [/opt/tiger/scripts/Aegisub/src/libresrc/libresrc.o] Error 1<br/>make: *** Waiting for unfinished jobs....<br/>At global scope:<br/>cc1plus: warning: unrecognized command line option "-Wno-c++11-narrowing"
</div>
</details>

So I installed the latest `wxWidgets` and build `Aegisub` success.

```
git clone https://github.com/Aegisub/Aegisub.git
cd Aegisub
apt install -y \
  autoconf libtool automake \
  libfreetype6-dev libfontconfig1-dev libass-dev \
  libboost-dev libboost-chrono-dev libboost-filesystem-dev \
  libboost-locale-dev libboost-regex-dev libboost-thread-dev \
  libffms2-dev libuchardet-dev libfftw3-dev \
  libice-dev \
  ffmpeg intltool
apt install libwxgtk3.0-dev libgtk-3-dev  -y
apt install lua5.1 libluajit-5.1-dev -y
autoreconf
./configure
make -j 4 # will failed, need to install wxWidgets

git clone https://github.com/wxWidgets/wxWidgets.git
cd wxWidgets/
git submodule update --init
./configure 
make -j 4

cd ..
./configure --with-wxdir=$PWD/wxWidgets
apt install luarocks -y
luarocks install busted
luarocks install moonscript
luarocks install uuid
make -j3 all test
cd automation
./aegisub-lua tests/busted.lua -p 'moon' tests/modules
```
